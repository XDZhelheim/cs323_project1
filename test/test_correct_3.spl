struct struct_correct_3 
{
  int x;
  char y;
};

int test_correct_3()
{
  struct struct_correct_3 object;
  object.x = object.y = 1;
}