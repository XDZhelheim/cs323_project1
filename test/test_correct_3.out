Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: struct_correct_3
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: x
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: char
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: y
                SEMI
          RC
      SEMI
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          TYPE: int
        FunDec (7)
          ID: test_correct_3
          LP
          RP
        CompSt (8)
          LC
          DefList (9)
            Def (9)
              Specifier (9)
                StructSpecifier (9)
                  STRUCT
                  ID: struct_correct_3
              DecList (9)
                Dec (9)
                  VarDec (9)
                    ID: object
              SEMI
          StmtList (10)
            Stmt (10)
              Exp (10)
                Exp (10)
                  Exp (10)
                    ID: object
                  DOT
                  ID: x
                ASSIGN
                Exp (10)
                  Exp (10)
                    Exp (10)
                      ID: object
                    DOT
                    ID: y
                  ASSIGN
                  Exp (10)
                    INT: 1
              SEMI
          RC
